src/zfp.c

```diff
@@ -766,7 +766,12 @@ zfp_stream_maximum_size(const zfp_stream* zfp, const zfp_field* field)
   maxbits += values - 1 + values * MIN(zfp->maxprec, zfp_field_precision(field));
   maxbits = MIN(maxbits, zfp->maxbits);
   maxbits = MAX(maxbits, zfp->minbits);
-  return ((ZFP_HEADER_MAX_BITS + blocks * maxbits + stream_word_bits - 1) & ~(stream_word_bits - 1)) / CHAR_BIT;
+  
+  uint64 maxsize = ((ZFP_HEADER_MAX_BITS + ((uint64)blocks) * maxbits + stream_word_bits - 1) & ~(stream_word_bits - 1)) / CHAR_BIT;
+
+  if (maxsize > SIZE_MAX)
```

Comment:
`SIZE_MAX` was introduced in C99.  We'll need a more general fix for C89.  Let me think about it.

Reply:
Yes, but the cleanest way would probably be to add
```
#ifndef SIZE_MAX
  #define SIZE_MAX ((size_t)-1)
#endif
```
at the end of [this](https://github.com/LLNL/zfp/blob/7d1e3a21047a976599b562b3bbd53b1f34348f1a/include/zfp/internal/zfp/types.h#L58) section.

Some compilers emit warnings when casting a signed literal to unsigned, so there may be variants like `((size_t)0 - 1)` that are needed to suppress such warnings.  I believe we have similar code elsewhere in zfp.  For now, the above may suffice to correct the bug.

Reply:
Though https://stackoverflow.com/a/44401992 suggests that some pre-processors don't support this? I don't know enough about C to know it that's correct so I'll go with your judgement